XIPs/xip-10-brandinfo-schema.md

```diff
@@ -0,0 +1,99 @@
+---
+xip: 10
+title: Brand info schema in conversation metadata
+description: Schema to enable standardized sharing of brand information as it relates to conversationId
+author: Yash Lunagaria (@yash-luna)
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-01-31
+---
+
+## Abstract
+
+This XIP proposes a schema for sharing brand information as it relates to `conversationId`. A standard way to represent this information in conversation `metadata` will enable different front-ends to surface the context captured by `conversationId` in a consistent, user-friendly manner while still affording clients design and UI flexibility.
+
+## Motivation
+
+The SDK API currently accepts a unique `conversationId` per address pair and allows `metadata` to be set as well. Today, different apps use these parameters in different ways and therefore other apps may not know how to render the `conversationId` appropriately within the UI to reflect the corresponding brand. We propose a schema to standardize the use of conversation `metadata` for the purpose of client-side brand expression stemming from the `conversationId`. The `metadata` field may include other non-standard (or future standard) properties.
+
+Note that the schema is only applicable when setting a non-null `conversationId`. If `conversationId` is null, the SDK does not allow setting `metadata`.
+
+## Specification
+
+Proposed `brandInfo` schema in conversation `metadata`
+
+```json
+{
+    conversationId: 'mydomain.xyz/abc/qrs',
+    metadata: {
+        brandInfo: {
+            displayName: 'My Company',
+            profileImage: 'mydomain.xyz/assets/myimage.png,
+            primaryColor: '#ffffff',
+        }
+    }
+}
+```
+
+Example `brandInfo` implementation for a chat app named Galaxy
+
+```json
+{
+    conversationId: 'galaxy.chat/dm/uniqueIdentifier',
+    metadata: {
+        brandInfo: {
+            displayName: 'Galaxy',
+            profileImage: 'galaxychat.com/assets/logo.png,
+            primaryColor: '#6865B8',
+        }
+    }
+}
+```
+
+The `profileImage` file is encouraged to meet the following criteria:
+
+- Aspect ratio: 1:1
+- Minimum resolution: 100x100
+- Maximum resolution: 800x800
+- Format: .PNG, .WEBP
+
+The `primaryColor` is a hex color code.
+
+## Backwards Compatibility
+
+Existing conversations will not be affected by the adoption of the new schema; only conversations created after the new schema is finalized will follow the updated schema.
+
+To maintain backwards compatibility, clients may consider rendering popular but older schemas (such as Lens), present in conversations created before the finalization of this XRC, in a manner consistent with their updated schema that follows this XRC.
+
+Old Lens schema:
+
+```json
+{
+        conversationId: lens.dev/dm/${memberA}-${memberB}
+        metadata: {}
+}
+```
+
+New Lens schema
+
+```json
+{
+    conversationId: 'lens.dev/dm/${memberA}-${memberB}',
+    metadata: {
+        brandInfo: {
```

Comment:
Metadata is currently restricted as a mapping that only accepts strings as values. We cannot have nested objects.

Reply:
- What was the rationale behind allowing strings only? 
- Can we relax restrictions and allow nested objects? If not, what's the best way to format the brandInfo as a string?

Reply:
> What was the rationale behind allowing strings only?

This is data that is going to be interpreted by many different apps who aren't going to be able to share a schema. If we left it free form, it opens up new avenues for client-side bugs (you start sending `brandinfo` as an object, some other app starts sending `brandinfo` as an array, and it's easy for one of your apps to break when it encounters a conversation from the other). I would think of this as something closer to HTTP headers than a free-form datastore. 

> Can we relax restrictions and allow nested objects? If not, what's the best way to format the brandInfo as a string?

It's baked into the proto schema, so it would be a significant change if we thought it was advisable (which I still don't).

One way to namespace the brandinfo would be to use dots in the keys. For example `brandinfo.displayName`. Could also embed a JSON object as `brandinfo`, but I'm less partial to that.